pyfr/integrators/dual/pseudo/pseudosteppers.py

```diff
@@ -337,3 +349,42 @@ class DualRK45PseudoStepper(DualRKVdH2RPseudoStepper):
     @property
     def _pseudo_stepper_order(self):
         return 4
+
+
+class DualDenseRKPseudoStepper(BaseDualPseudoStepper):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        scheme = resource_string('pyfr.integrators', 'schemes/' + self.path)
+
+        self.a, self.b = _get_coefficients(scheme.decode())
+
+    def step(self, t):
+        add = self._add
+        rhs = self._rhs_with_dts
+
+        r = self._pseudo_stepper_regidx
+
+        # Ensure r0 references the bank containing u(n+1,m)
+        if r[0] != self._idxcurr:
+            r[0], r[1] = r[1], r[0]
```

Comment:
Do we want to be messing with the entries of _pseudo_stepper_regidx?  In other code we just swap variables, but here I think we are actually operating on _pseudo_stepper_regidx which makes me uncomfortable.  Perhaps wrap L366 with list(...)?

Reply:
I don't think we are modifying anything, since

    @property
    def _pseudo_stepper_regidx(self):
        return self._regidx[:self._pseudo_stepper_nregs]

r = self._pseudo_stepper_regidx
Just copies all pseudo stepper register indices from the self._regidx list that contains all register indices.

Reply:
So that's an implementation detail in the sense that we are implicitly relying on a copy being made.  However, should _pseudo_stepper_regidx become a list in its own right or if self._regidx should be changed from a list to a NumPy array of ints this would no longer be the case.  Neither of these are things which one would expect to break code, but in this case it would.